---
title: "HbF Genomics"
subtitle: "Codebook: Novel Fetal Hemoglobin Modifying Loci Revealed with Genome Wide Studies of Sickle Cell Disease Patients from Cameroon, and Global Metanalysis"
author: 
  - "Kevin Esoh (eshkev001@myuct.ac.za)"
  - "Jack Morrice ()"
date: "September 4, 2022"
output: 
  html_document:
    number_sections: TRUE
    #toc: TRUE
    #toc_depth: 2
    #toc_float: TRUE
    df_print: paged
    theme: default
    highlight: tango
knit: (function(inputFile=list.files('.', pattern="*.rmd")) {
  rmarkdown::render(inputFile, output_dir = "~/esoh/data/sadaccgwas/writeup/html") })
tags: [jekyll, r-markdown]
always_allow_html: TRUE
---

```{r, include = TRUE, echo = TRUE, results='hide', message=FALSE}
library(tidyverse)
```

```{r global_options, include = TRUE, echo = FALSE}
knitr::opts_chunk$set(message=FALSE,
tidy.opts=list(width.cutoff=60))
```

```{r, include = FALSE, echo = FALSE}
# Control tabbed sections
# {.tabset .tabset-fade .tabset-pills}
```

# Clinical datasets {.tabset .tabset-fade .tabset-pills}
```{r}
cm <- read_table("../clinicaldata/cm_clinical_data.tsv")
tz <- read_table("../clinicaldata/tz_clinical_data.tsv")
```

# Summary of Cameroon data
## variables
```{r}
cm %>% names()
```

## Summary statistics
### Quantitative/Continuous variables
```{r}
variables <- c(
   "AGE",					# Age in years
   "HbS", 					# Proportion of HbS
   "HbF",					# Proportion of fetal haemoglobin
   "HbA2_percentage",		# Proportion of HbA2
   "Consultation.year",		# Number of consultations per year
   "MCHC",					# Mean corpuscular haemoglobin concentration
   "Mono",					# Monocyte count (x10^9/l)
   "Hosp",					# Number of hospitalizations per year
   "RBC",					# Red blood cell count (x10^12/l)
   "WBC",					# White blood cell count (x10^9/l)
   "Platelets",				# Platelet count (x10^9/l)
   "VOC.year",				# Number of vaso occlusive crises per year
   "MCV.fl.",				# Mean cell volume
   "Lymp"					# Lymphocyte count (x10^9/l)
)
cm %>% select(variables) %>% summary()
```

### Categorigal variables
#### Sex
```{r, fig.align = 'center', fig.cap = "Figue 1. Proportion of females and males in the discovery dataset of Cameroonian inidividuals"}
cm %>% 
   select(SEX) %>% 
   table() %>% 
   as.data.frame() %>% 
   na.omit() %>%
   mutate(Prop = (Freq/sum(Freq))*100) -> sex		# Sex

sex
colnames(sex) <- c("cat", "freq", "prop")
par(mar=c(4,4,3,4))
barplot(
   sex$prop, 
   ylim=c(0, 60), 
   col=c(2,4)
)
text(
   x=c(0.7,1.9), 
   y=30, 
   labels=c("Female", "Male")
)
mtext(
   text = "Proportion (%)",
   side = 2,
   line = 3
)
mtext(
   text = "Sex",
   side = 1,
   line = 1
)
```

#### Number of patients with at least one overt stroke
```{r}
cm %>% select(Stroke) %>% table()               # Number of patients with at least one overt stroke
```

#### 3.7 kb HbA1/HbA2 alpha thalassaemia deletion
```{r}
cm %>% select(athal) %>% table()		# 3.7 kb HbA1/HbA2 alpha thalassaemia deletion
```

#### HBB gene cluster haplotypes
```{r}
cm %>% select(Haplotypes) %>% table()		# HBB gene cluster haplotypes
```
*CAM = CAM/CAM, BEN = BEN/BEN, SEN = SEN/SEN, CAR = CAR/CAR, Atypical = Atypical/Atypical*


# Summary of Tanzania data
## Variables
```{r}
tz %>% names()
```

## Summary statistics
### Quantitative/Continuous variables
```{r}
variables <- c(
   "AGE",
   "HbF"
)
tz %>% select(variables) %>% summary()
```

### Categorical variable
#### Sex
```{r}
tz %>%
   select(SEX) %>%
   table() %>%
   head()
```
Excluding individuals with ambiguous sex information and recoding
```{r, fig.align = 'center', fig.cap = "Figure 2. Proportion of females and males in the replication dataset of Tanzanian inidividuals"}
tz %>%
   mutate(sex.updated = ifelse(SEX == 1, "M", ifelse(SEX == 2, "F", NA))) %>%
   select(sex.updated) %>%
   table() %>%
   as.data.frame() %>%
   na.omit() %>%
   mutate(Prop = (Freq/sum(Freq))*100) -> sex           # Sex

sex
colnames(sex) <- c("cat", "freq", "prop")
par(mar=c(4,4,3,4))
barplot(
   sex$prop,
   ylim=c(0, 60),
   col=c(2,4)
)
text(
   x=c(0.7,1.9),
   y=30,
   labels=c("Female", "Male")
)
mtext(
   text = "Proportion (%)",
   side = 2,
   line = 3
)
mtext(
   text = "Sex",
   side = 1,
   line = 1
)
```

---------

# HbF normalization (Cubic root)

## Cameroon
```{r}
cm %>% 
   mutate(normalizedHbF = HbF^(1/3)) -> cm.hbf.norm 
   
write.table(
   cm.hbf.norm, 
   "../clinicaldata/cm_clinical_data_updated.tsv",
   col.names = T,
   row.names = F,
   quote = F,
   sep = "\t"
)

cm.hbf.norm %>% 
   select(HbF, normalizedHbF) %>% 
   head()
```

## Tanzania
```{r}
# Tanzania
tz %>%
   mutate(normalizedHbF = HbF^(1/3)) -> tz.hbf.norm

write.table(
   tz.hbf.norm,
   "../clinicaldata/tz_clinical_data_updated.tsv",
   col.names = T,
   row.names = F,
   quote = F,
   sep = "\t"
)

tz.hbf.norm %>% 
   select(HbF, normalizedHbF) %>% 
   head()
```

## HbF distribution before and after normalization
```{r, fig.show = "hold", fig.align = 'center', fig.cap = "Figure 3. HbF distribution in Cameroonian and Tanzanian study participants before and after normalization", out.width = "50%"}
cm.hbf.den <- density(cm.hbf.norm$HbF)
cm.nhbf.den <- density(cm.hbf.norm$normalizedHbF)
tz.hbf.den <- density(tz.hbf.norm$HbF)
tz.nhbf.den <- density(tz.hbf.norm$normalizedHbF)

#par(mfrow=c(2,2))
par(mar=c(4,4,3,4))
plot(cm.hbf.den, main="Cameroon HbF")
polygon(cm.hbf.den, col=2)
plot(cm.nhbf.den, ylab="", main="Cameroon normalized HbF")
polygon(cm.nhbf.den, col=5)
plot(tz.hbf.den, main="Tanzania HbF")
polygon(tz.hbf.den, col=2)
plot(tz.nhbf.den, ylab="", main="Tanzania normalized HbF")
polygon(tz.nhbf.den, col=5)
```
